### ISSUE TITLE
warn / error flags to cargo scout ?

### ISSUE THREAD

**Issue by o0Ignition0o** [o0Ignition0o/cargo-scout#26]
in order to be able to use cargo-scout programmatically, it might be nice to have it warn when lints are found, and not error out.

It would be put behind a flag

**Comment by daubaris** [comment]
Currently it display the errors and the count of the errors if any were found. Do you mean to add a flag which just outputs the count of the errors?

**Comment by cbonaudo** [comment]
I think what @o0Ignition0o meant is that when the lints are found, cargo-scout exits with an error.
We might not always need it to error, maybe just warn the user (by showing them the lint as a warning).
For example if they don't want to break their CI. 
And if they want so, they can use a --warn or some sort of flag to not get warning and not errors.
```
Clippy::pedantic found 3 warnings
Error: NotClean
error: process didn't exit successfully: `target\debug\cargo-scout.exe` (exit code: 1)
```

**Comment by o0Ignition0o** [comment]
Hi, exactly !
When you run a command on Linux or mac OS, you get an exit code, which tells you if everything ran well or not.
You can get the exit code by typing `echo $?` after running a command, as shown here:

<img width="459" alt="image" src="https://user-images.githubusercontent.com/9465678/70372767-b74ad900-18e3-11ea-9212-5cc7691c4ca8.png">

An exit code different from 0 is considered to mean an error occured.

The goal of this issue would be to have a flag that tells cargo-scout if it should throw an error as done [here in the code](https://github.com/o0Ignition0o/cargo-scout/blob/master/src/main.rs#L79), or return a [success code](https://github.com/o0Ignition0o/cargo-scout/blob/master/src/main.rs#L72) and just display the lints that have been found.

Does it make sense ? Please let me know if there's any question :)

**Comment by daubaris** [comment]
Oh, okay. So basically add two flags, one for exiting with an error status and another one exiting without it and displaying the lints in both cases, right?

**Comment by o0Ignition0o** [comment]
Displaying the lints in both cases is definitely something we want to do :)
As for the flags, I would say we add only one flag, "warn-only" so we can keep erroring out as default.
"warn-only" doesn't sound too good, maybe we can come up with a better name though ^^

**Comment by daubaris** [comment]
I'd like to work on this if you don't mind.

**Comment by o0Ignition0o** [comment]
Sure, thank's a lot for stepping in ! If you have any questions please let me know :D

**Comment by o0Ignition0o** [comment]
Fixed in #35 !
